### PR TITLE
Replacing AvatarView with Avatar Vnext throughout FluentUI library.

### PR DESCRIPTION
**Pull request by rdeassis**
### Platforms Impacted
- [x] iOS
- [ ] macOS

### Description of changes

This change replaces all uses of the current AvatarView (written on UIKit) with the SwiftUI Avatar Vnext throughout the Library.
The AvatarGroupView is not being updated as it'll be re-written in SwiftUI.
#517, #527 and #528 unblocked key scenarios to make this update possible.

### Verification

https://user-images.githubusercontent.com/68076145/115629482-45e0b000-a2b7-11eb-9602-15406ea8d552.mov

Demo app scenarios on the following controls:

**- ContactCollectionView**

| iPhone                                       | iPad                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 11 Pro - 2021-04-21 at 15 06 08 | Simulator Screen Shot - iPad Pro (12 9-inch) (4th generation) - 2021-04-21 at 15 07 30 |

**- NavigationController**

| iPhone                                       | iPad                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 12 - 2021-04-21 at 15 09 12 | Simulator Screen Shot - iPad Pro (12 9-inch) (4th generation) - 2021-04-21 at 15 09 34 |

**- PeoplePicker**

| iPhone                                       | iPad                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 12 - 2021-04-21 at 15 11 30 | Simulator Screen Shot - iPad Pro (12 9-inch) (4th generation) - 2021-04-21 at 15 10 33 |

**- PersonaListView**

| iPhone                                       | iPad                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 11 Pro - 2021-04-21 at 15 12 29 | Simulator Screen Shot - iPad Pro (12 9-inch) (4th generation) - 2021-04-21 at 15 12 52 |

**- SideTabBar**

| iPhone                                       | iPad                                      |
|----------------------------------------------|--------------------------------------------|
| Simulator Screen Shot - iPhone 11 Pro - 2021-04-21 at 15 14 08 | Simulator Screen Shot - iPad Pro (12 9-inch) (4th generation) - 2021-04-21 at 15 14 24 |

### Pull request checklist

This PR has considered:
- [x] Light and Dark appearances
- [x] VoiceOver and Keyboard Accessibility
- [ ] Internationalization and Right to Left layouts
- [ ] Different resolutions (1x, 2x, 3x)
- [x] Size classes and window sizes (iPhone vs iPad, notched devices, multitasking, different window sizes, etc)

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/microsoft/fluentui-apple/pull/535)

### CODE REVIEW

File: ios/FluentUI.Demo/FluentUI.Demo/Demos/SideTabBarDemoController.swift [hunk 1]
```diff
@@ -184,14 +184,14 @@ class SideTabBarDemoController: DemoController {
     }
 
     private func showAvatarView(_ show: Bool) {
-        var avatarView: AvatarLegacyView?
+        var avatar: MSFAvatar?
         if let image = UIImage(named: "avatar_kat_larsson"), show {
-            avatarView = AvatarLegacyView(avatarSize: .medium, withBorder: false, style: .circle, preferredFallbackImageStyle: .onAccentFilled)
-            avatarView?.setup(primaryText: "Kat Larson", secondaryText: "", image: image)
-            avatarView?.hasPointerInteraction = true
```
Comment (sophialee0416) [inline ios/FluentUI.Demo/FluentUI.Demo/Demos/SideTabBarDemoController.swift:191]: If we added pointer interaction, do we want to include this too?

Comment (rdeassis) [reply ios/FluentUI.Demo/FluentUI.Demo/Demos/SideTabBarDemoController.swift:191]: Thanks for catching it. I'll set it for the side tab bar too.

File: ios/FluentUI/People Picker/ContactView.swift [hunk 2]
```diff
@@ -21,12 +21,12 @@ open class ContactView: UIControl {
         case large
         case small
 
-        var avatarSize: AvatarLegacySize {
+        var avatarSize: MSFAvatarSize {
             switch self {
             case .large:
-                return .extraExtraLarge
+                return .xxlarge
```
Comment (sophialee0416) [inline ios/FluentUI/People Picker/ContactView.swift:27]: Nit: capitalize the L in large to reflect Hunter's changes (#514)?

**Review by sophialee0416 — APPROVED** [review]

File: ios/FluentUI/People Picker/ContactView.swift (hunk 2 above)
Comment (rdeassis) [reply ios/FluentUI/People Picker/ContactView.swift:27]: Thanks! I'll do that in a separate PR to normalize all of these.

**Pull request merged**